plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1236 @@
+    {
+        if (!hasPermission(createResource(entity), context, RangerTrinoAccessType.GRANT, "GrantEntityPrivilege")) {
+            AccessDeniedException.denyGrantEntityPrivilege(privilege.toString(), entity);
+        }
+    }
+
+    @Override
+    public void checkCanDenyEntityPrivilege(SystemSecurityContext context, EntityPrivilege privilege, EntityKindAndName entity, TrinoPrincipal grantee)
+    {
+        if (!hasPermission(createResource(entity), context, RangerTrinoAccessType.REVOKE, "DenyEntityPrivilege")) {
+            AccessDeniedException.denyDenyEntityPrivilege(privilege.toString(), entity);
+        }
+    }
+
+    @Override
+    public void checkCanRevokeEntityPrivilege(SystemSecurityContext context, EntityPrivilege privilege, EntityKindAndName entity, TrinoPrincipal revokee, boolean grantOption)
+    {
+        if (!hasPermission(createResource(entity), context, RangerTrinoAccessType.REVOKE, "RevokeEntityPrivilege")) {
+            AccessDeniedException.denyRevokeEntityPrivilege(privilege.toString(), entity);
+        }
+    }
+
+    @Override
+    public void checkCanCreateRole(SystemSecurityContext context, String role, Optional<TrinoPrincipal> grantor)
+    {
+        if (!hasPermission(createRoleResource(role), context, RangerTrinoAccessType.CREATE, "CreateRole")) {
+            AccessDeniedException.denyCreateRole(role);
+        }
+    }
+
+    @Override
+    public void checkCanDropRole(SystemSecurityContext context, String role)
+    {
+        if (!hasPermission(createRoleResource(role), context, RangerTrinoAccessType.DROP, "DropRole")) {
+            AccessDeniedException.denyDropRole(role);
+        }
+    }
+
+    @Override
+    public void checkCanShowRoles(SystemSecurityContext context)
+    {
+        if (!hasPermission(createRoleResource("*"), context, RangerTrinoAccessType.SHOW, "ShowRoles")) {
+            AccessDeniedException.denyShowRoles();
+        }
+    }
+
+    @Override
+    public void checkCanGrantRoles(SystemSecurityContext context, Set<String> roles, Set<TrinoPrincipal> grantees, boolean adminOption, Optional<TrinoPrincipal> grantor)
+    {
+        if (!hasPermission(createRoleResources(roles), context, RangerTrinoAccessType.GRANT, "GrantRoles")) {
+            AccessDeniedException.denyGrantRoles(roles, grantees);
+        }
+    }
+
+    @Override
+    public void checkCanRevokeRoles(SystemSecurityContext context, Set<String> roles, Set<TrinoPrincipal> grantees, boolean adminOption, Optional<TrinoPrincipal> grantor)
+    {
+        if (!hasPermission(createRoleResources(roles), context, RangerTrinoAccessType.REVOKE, "RevokeRoles")) {
+            AccessDeniedException.denyRevokeRoles(roles, grantees);
+        }
+    }
+
+    @Override
+    public void checkCanShowCurrentRoles(SystemSecurityContext context)
+    {
+        //allow
```

Comment:
Why is this allowed by default? Doesn't Ranger have roles?

Reply:
Ranger does have roles. This method returns if it is okay for Trino to show current roles of the user. I am not aware of cases where a user is not allowed to see their current roles. Hence allowing the user to view their current roles.